fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -2031,79 +2101,34 @@ private TransactionSynchronizationRegistry getTrxSynchRegistry() throws FHIRPers
+                        for (ExtractedParameterValue p : parameters) {
+                            if (wholeSystemParam) {
+                                p.setWholeSystem(true);
                             }
-                        } catch (IllegalArgumentException e) {
-                            // log and continue with the other parameters
-                            StringBuilder msg = new StringBuilder("Skipping search parameter '" + code + "'");
-                            if (sp.getId() != null) {
-                                msg.append(" with id '" + sp.getId() + "'");
-                            }
-                            msg.append(" for resource type " + fhirResource.getClass().getSimpleName());
-                            // just use the message...no need for the whole stack trace
-                            msg.append(" due to \n" + e.getMessage());
+                            extractedParameters.getParameters().add(p);
                             if (log.isLoggable(Level.FINE)) {
-                                log.fine(msg.toString());
+                                log.fine("Extracted Parameter '" + p.getName() + "' from Resource.");
                             }
-                            addWarning(IssueType.INVALID, msg.toString());
-                        }
-                    }
-                    // retrieve the list of parameters built from all the FHIRPathElementNode values
-                    List<ExtractedParameterValue> parameters = parameterBuilder.getResult();
-                    for (ExtractedParameterValue p : parameters) {
-                        if (wholeSystemParam) {
-                            p.setWholeSystem(true);
-                        }
-                        allParameters.add(p);
-                        if (log.isLoggable(Level.FINE)) {
-                            log.fine("Extracted Parameter '" + p.getName() + "' from Resource.");
                         }
                     }
                 }
+
+                // Augment the extracted parameter list with special values we use to represent compartment relationships.
+                // These references are stored as tokens and are used by the search query builder
+                // for compartment-based searches
+                addCompartmentParams(extractedParameters.getParameters(), fhirResource);
             }
 
-            // Augment the extracted parameter list with special values we use to represent compartment relationships.
-            // These references are stored as tokens and are used by the search query builder
-            // for compartment-based searches
-            addCompartmentParams(allParameters, fhirResource);
+            // Generate the hash which is used to quickly determine whether the extracted parameters
+            // are different than the extracted parameters that currently exist in the database
+            extractedParameters.generateHash(parameterHashUtil);
```

Comment:
Perhaps ExtractedSearchParameters should be immutable like a DTO. Just construct one at the end with the list of parameters and the computed hash. I don't think ExtractedSearchParameters should be in the business of computing the hash itself, just holding it.

Reply:
Updated to be immutable.